src/main/java/greencity/service/impl/PlaceServiceImpl.java

```diff
@@ -149,41 +118,37 @@ public Place update(PlaceUpdateDto dto) {
         placeRepo.save(updatedPlace);
 
         updateOpening(dto.getOpeningHoursList(), updatedPlace);
-        updateDiscount(dto.getDiscounts(), updatedCategory, updatedPlace);
+        updateDiscount(dto.getDiscounts(), updatedPlace);
 
         return updatedPlace;
     }
 
     /**
-     * Method for updating set of {@link Discount} and save with new {@link Category} and {@link Place}.
+     * Method for updating set of {@link DiscountValue} and save with new {@link Category} and {@link Place}.
      *
-     * @param discountDtos    - set of {@link Discount}.
-     * @param updatedCategory - {@link Category} entity.
-     * @param updatedPlace    - {@link Place} entity.
+     * @param discounts    - set of {@link DiscountValue}.
+     * @param updatedPlace - {@link Place} entity.
      * @author Kateryna Horokh
      */
-    private void updateDiscount(Set<DiscountDto> discountDtos, Category updatedCategory, Place updatedPlace) {
+    private void updateDiscount(Set<DiscountValueDto> discounts, Place updatedPlace) {
         log.info(LogMessage.IN_UPDATE_DISCOUNT_FOR_PLACE);
 
-        Set<Discount> discountsOld = discountService.findAllByPlaceId(updatedPlace.getId());
+        Set<DiscountValue> discountsOld = discountService.findAllByPlaceId(updatedPlace.getId());
         discountService.deleteAllByPlaceId(updatedPlace.getId());
-        Set<Discount> discounts = new HashSet<>();
-        discountDtos.forEach(d -> {
-            Discount discount = modelMapper.map(d, Discount.class);
-            Specification specification = specificationService.findByName(d.getSpecification().getName());
+        Set<DiscountValue> newDiscounts = new HashSet<>();
+        discounts.forEach(d -> {
+            DiscountValue discount = discountValueMapper.convertToEntity(d);
             discount.setPlace(updatedPlace);
-            discount.setCategory(updatedCategory);
-            discount.setSpecification(specification);
             discountService.save(discount);
-            discounts.add(discount);
+            newDiscounts.add(discount);
         });
-        discountsOld.addAll(discounts);
+        discountsOld.addAll(newDiscounts);
```

Comment:
Why you add to discountsOld newDiscounts, but nothing do later with discountsOld? And what you do with newDiscounts? You only add it to variable discountsOld and then method ends.